homeassistant/components/tahoma/manifest.json

```diff
@@ -10,5 +10,11 @@
     "@imicknl",
     "@vlebourl",
     "@tetienne"
+  ],
+  "dhcp": [
+    {
+      "hostname": "gateway*",
+      "macaddress": "F8811A*"
```

Comment:
See https://ouilookup.com/vendor/overkiz. We cannot use Zeroconf (like Somfy integration), since it will only broadcast that service when the device has HomeKit support. Many Overkiz hubs that are supported with this integration don't support HomeKit.

Reply:
dhcp is going to use the default unique id of `DOMAIN` which means its likely going to get unexpectedly rediscovered.

Would could implement a `async_step_dhcp` that aborts if `self._async_current_entries(include_ignore=False)` is true and otherwise passes to `async_step_user`

Reply:
I was waiting for https://github.com/home-assistant/core/pull/46921, since the gateway id is part of the hostname, thus we could add a `async_step_dhcp` that checks if their is a config entry with the same gateway id.

I thought other integrations also had this DHCP option, without having the `async_step_dhcp` implemented, but happy to take it out here and implement the dhcp with gateway detection in a follow up PR.

Reply:
I'm wrong on this one `_async_handle_discovery_without_unique_id` actually already handles this case ```

        # Abort if the handler has config entries already
        if self._async_current_entries():
            raise data_entry_flow.AbortFlow("already_configured")

```

So this good as is.  Sorry for leading down the rabbit hole.